torchaudio/csrc/ffmpeg/prototype.cpp

```diff
@@ -1,402 +1,129 @@
 #include <torch/script.h>
-#include <torchaudio/csrc/ffmpeg/streamer.h>
+#include <torchaudio/csrc/ffmpeg/stream_reader_wrapper.h>
 #include <stdexcept>
 
 namespace torchaudio {
 namespace ffmpeg {
 
 namespace {
 
-using OptionDict = c10::Dict<std::string, std::string>;
-
-std::map<std::string, std::string> convert_dict(
-    const c10::optional<OptionDict>& option) {
-  std::map<std::string, std::string> opts;
-  if (option) {
-    for (auto& it : option.value()) {
-      opts[it.key()] = it.value();
-    }
+OptionDict map(const c10::optional<c10::Dict<std::string, std::string>>& dict) {
+  OptionDict ret;
+  if (!dict.has_value()) {
+    return ret;
+  }
+  for (const auto& it : dict.value()) {
+    ret.insert({it.key(), it.value()});
   }
-  return opts;
+  return ret;
 }
 
-struct StreamerHolder : torch::CustomClassHolder {
```

Comment:
The implementation of `StreamHolder` has been moved to StreamReaderBinding in `stream_reader_wrapper.[h|cpp]`.